### PR TITLE
Create sub-modules for examples and wrappers

### PR DESCRIPTION
**Pull request by rs**

### CODE REVIEW

File: wrapper/go.mod [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module github.com/rs/cors/wrapper/gin
```
Comment (rs) [inline wrapper/go.mod]: @komuw: shouldn't this file be at `wrapper/gin` level?

Comment (komuw) [reply wrapper/go.mod]: @rs agreed. 
I have done that now.

**Comment by komuw** [comment]
@rs I do not know if you want one `go.mod` for all the examples(like I've done) or one per example directory?

**Comment by rs** [comment]
When running `go test` in `wrapper/gin`, I still get this error:

```
go: finding golang.org/x/net v0.0.0-20181003013248-f5e5bdd77824
can't load package: package github.com/rs/cors/wrapper/gin: unknown import path "github.com/rs/cors/wrapper/gin": ambiguous import: found github.com/rs/cors/wrapper/gin in multiple modules:
	github.com/rs/cors/wrapper/gin (/Users/rs/Development/cors/wrapper/gin)
	github.com/rs/cors v1.5.0 (/Users/rs/go/pkg/mod/github.com/rs/cors@v1.5.0/wrapper/gin)
```

How do you test this?

**Comment by komuw** [comment]
@rs 
after this commit: https://github.com/rs/cors/pull/67/commits/6cabe93fa7e901bc68a02f4dee3b8f3aaeb9f96e

the commands:
```bash
go test -count=1 ./... && \
cd wrapper/gin && \
go test -count=1 ./... && \
cd ../..
```
```sh
ok  	github.com/rs/cors	0.004s
ok  	github.com/rs/cors/wrapper/gin	0.004s
```
now works.

However the command;
```bash
cd examples && \
go test -count=1 ./... && \
cd ..
```
doesn't succeed. 
The only way I can make the last command succeed is if I remove;
https://github.com/rs/cors/blob/9a47f48565a795472d43519dd49aac781f3034fb/examples/gin/server.go#L7 
which seems to indicate that maybe the replace directive: https://github.com/rs/cors/pull/67/files#diff-8d6f6299a802b70c5794f4ed12c66ce2R26
is not working. 

To be honest, this has turned out quite harder than I thought it would. 
I think(more like hoping), that the issue; https://github.com/golang/go/issues/27056 is meant to address this. 

I'll keep probing to see if I can get the examples folder to play cool

**Comment by komuw** [comment]
also: https://github.com/golang/go/issues/27957

**Comment by komuw** [comment]
this investigation  has led me to discover something else;
it seems like Go modules doesn't verify all replace directives. 
I have filed; https://github.com/golang/go/issues/28032 for that

**Pull request closed without merging**